sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2494,16 +2494,18 @@ class Dataset[T] private[sql](
    * @since 2.0.0
    */
   def toJSON: Dataset[String] = {
-    val rowSchema = this.schema
-    val rdd: RDD[String] = queryExecution.toRdd.mapPartitions { iter =>
+    import sparkSession.implicits.newStringEncoder
```

Comment:
The encoder is for the implicit mapPartitions argument. We need to have Encoder[String] for the output. Will put it just above the mappartitions call